sklearn/model_selection/tests/test_search.py

```diff
@@ -242,6 +249,32 @@ def test_grid_search_pipeline_steps():
     assert not hasattr(param_grid["regressor"][1], "coef_")
 
 
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype", yield_namespace_device_dtype_combinations()
+)
+@pytest.mark.parametrize("SearchCV", [GridSearchCV, RandomizedSearchCV])
+def test_array_api_SearchCV(SearchCV, array_namespace, device, dtype):
+    xp = _array_api_for_tests(array_namespace, device)
+
+    X = np.arange(100).reshape((10, 10))
+    X_np = X.astype(dtype)
+    X_xp = xp.asarray(X_np, device=device)
+
+    # y should always be an integer, no matter what `dtype` is
+    y_np = np.array([0] * 5 + [1] * 5)
+    y_xp = xp.asarray(y_np, device=device)
+
+    with config_context(array_api_dispatch=True):
+        searcher = SearchCV(
+            LinearDiscriminantAnalysis(),
+            {"tol": [1e-2, 1e-3, 1e-4, 1e-5, 1e-6, 1e-7]},
+            cv=2,
+        )
+        searcher.fit(X_xp, y_xp)
+
+        searcher.score(X_xp, y_xp)
```

Comment:
Note sure about the value of this test. It seems redundant with our `check_array_api_input` triggered by `test_search_cv` when the underlying estimator accepts array API inputs (see 0e3c7d7).

This seess to work as expected (at least on my local dev host), for instance:

```
sklearn/tests/test_common.py::test_search_cv[GridSearchCV(cv=2,error_score='raise',estimator=Ridge(),param_grid={'alpha':[0.1,1.0]})-check_array_api_input(array_namespace=torch,dtype_name=float32,device=cpu)] PASSED [  5%]
```

Reply:
Agreed. I think I added it way back when we didn't have the common test yet.